### PR TITLE
Fix wifi for Aura One

### PR DESCRIPTION
**Pull request by houqp**
Aura One uses 8189fs instead of dhd module. Depends on https://github.com/koreader/koreader-base/pull/476.

### CODE REVIEW

File: platform/kobo/enable-wifi.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@
 # Load wifi modules and enable wifi.
 
 lsmod | grep -q sdio_wifi_pwr || insmod /drivers/$PLATFORM/wifi/sdio_wifi_pwr.ko
-lsmod | grep -q dhd || insmod /drivers/$PLATFORM/wifi/dhd.ko
+# WIFI_MODULE_PATH = /drivers/$PLATFORM/wifi/$WIFI_MODULE.ko
+lsmod | grep -q $WIFI_MODULE || insmod $WIFI_MODULE_PATH
```
Comment (Frenzie) [inline platform/kobo/enable-wifi.sh]: Will this be set on older firmware or should there be a check for an empty $WIFI_MODULE var?

Comment (houqp) [inline platform/kobo/enable-wifi.sh]: Anyone using KSM should have this one set. Based on the comment in KSM, it seems like a variable set by the default init script just like `PLATFORM`.

**Comment by houqp** [comment]
OK, so it's failing because wpa_supplicant doesn't accept my set password request, weird :(

**Comment by houqp** [comment]
OK. So looks like A1's wpa_supplicant is compiled with CONFIG_NO_PBKDF2, which means support for  ASCII passphrase is disabled.

File: platform/kobo/enable-wifi.sh (hunk 1 above)
Comment (Frenzie) [inline platform/kobo/enable-wifi.sh]: Well yes, but I don't think that necessarily means it was already set, say, five years ago. To partially answer my own question, this is in the /etc/init.d/rcS from when I first got my H2O back in 2014:

```bash
INTERFACE=wlan0
WIFI_MODULE=ar6000
if [ $PLATFORM != freescale ]; then
	INTERFACE=eth0
	WIFI_MODULE=dhd
fi

export PLATFORM
export PRODUCT
export INTERFACE
export WIFI_MODULE

export NICKEL_HOME=/mnt/onboard/.kobo
export LD_LIBRARY_PATH=/usr/local/Kobo
export WIFI_MODULE_PATH=/drivers/$PLATFORM/wifi/$WIFI_MODULE.ko
export LANG=en_US.UTF-8
```

A quick search shows that the relevant variable was already there in 2012: https://github.com/matsuu/KoboRoot/blob/1c648a040c3afe1899a22e8e2d4357d395e5b46c/etc/init.d/rcS#L97-L102

So I can stop worrying. ;-)

**Comment by houqp** [comment]
@dset0x I was under the impression that only newer models ship with this driver. Then it's very likely that new firmware on old model also ships with CONFIG_NO_PBKDF2 in wpa_supplicant, which makes koreader unable to set network credentials. I am still debating on whether we should reimplement the sha1 mac for PSK in Lua or port over hostapd's C implementation. We can also shell out to run wap_passphrase as a quick hack.

@dset0x can you confirm that you have WIRELESS_MODULE in your /etc/init.d/rcS set as 8189fs?

**Comment by dset0x** [comment]
@houqp Here is my [rcS](https://gist.github.com/dset0x/69d5507ff7a898127544ad0b565c6cc3) and here's what it sets:

```
PLATFORM: mx6sl-ntx
WIFI: WC121A2
WIFI_MODULE: dhd
```

The latest changes on my box are a full upgrade to 4.2.8432, followed by a re-installation of KSM8 and then applying the rcS update provided by tshering [here](https://www.mobileread.com/forums/showpost.php?p=3389190&postcount=221).

**Comment by houqp** [comment]
OOPS, my fix in this PR won't work then since KSM still ships with the old module name :( Something we should following with tshering. At the mean time, we can make our wifi enable script smarter.

**Comment by houqp** [comment]
@dset0x , hmm... I am confused now. are you able to run `insmod /drivers/mx6sl-ntx/wifi/dhd.ko` without error? I assume 8189fs is not for your model?

**Comment by dset0x** [comment]
@houqp, correct. Upon `ssh`ing while already on WiFi:
```
[root@(none) root]# lsmod
sdio_wifi_pwr 486 0 - Live 0x7f229000
dhd 242221 0 - Live 0x7f1de000
[root@(none) root]# insmod /drivers/mx6sl-ntx/wifi/dhd.ko
insmod: can't insert '/drivers/mx6sl-ntx/wifi/dhd.ko': File exists
```

**Comment by houqp** [comment]
@dset0x thanks for the test. So two things:

1. This PR won't break the wifi enable script for you since you get the correct WIFI_MODULE in your env var
2. This PR won't fix your wifi issue, since you are already loading dhd.ko. we might still need to add `-Dwext`. Can you confirm this is the only change needed to make wifi work for you?

Another thing you can check is to enable wifi in nickel and in your ssh shell find out which wifi module it loaded and what arguments it passed to wpa_supplicant.

**Comment by dset0x** [comment]
@houqp 

> Can you confirm this is the only change needed to make wifi work for you?

Yes.

> Another thing you can check is to enable wifi in nickel and in your ssh shell find out which wifi module it loaded and what arguments it passed to wpa_supplicant.

Good idea,
`wpa_supplicant -D wext -s -i eth0 -c /etc/wpa_supplicant/wpa_supplicant.conf -C /var/run/wpa_supplicant -B`

FWIW, I think that "Scanning" takes longer since the update, but I could be just imagining it.

**Comment by Hzj-jie** [comment]
FYI, wpa_supplicant on Aura HD also does not function after 4.2 stock os update.

**Comment by houqp** [comment]
@Hzj-jie can you try with the ` -D wext` fix? Also can you confirm that it's still using dhd from output of `lsmod`?

**Comment by houqp** [comment]
@Frenzie @Hzj-jie tested and verified with the latest firmware on my A1. This patch set depends on https://github.com/koreader/koreader-base/pull/476.

**Pull request merged**